### PR TITLE
Add more packages to pinned version policy

### PR DESCRIPTION
**Pull request by timotheeguerin**

### CODE REVIEW

**Review by nguerrera — APPROVED** [review]

File: rush.json [hunk 1]
```diff
@@ -154,19 +154,19 @@
...
       "projectFolder": "packages/eslint-config-typespec",
       "reviewCategory": "production",
-      "shouldPublish": true
+      "versionPolicyName": "typespec"
     },
     {
       "packageName": "@typespec/internal-build-utils",
       "projectFolder": "packages/internal-build-utils",
       "reviewCategory": "production",
-      "shouldPublish": true
```
Comment (nguerrera) [inline rush.json:163]: wait these weren't published before?

Comment (nguerrera) [reply rush.json:163]: I guess we'd have to publish it at some point to get rid of submodule.

Comment (nguerrera) [reply rush.json:163]: commented on wrong line, spec wasn't published before.

File: rush.json [hunk 2]
```diff
@@ -154,19 +154,19 @@
...
+      "versionPolicyName": "typespec"
     },
     {
       "packageName": "@typespec/spec",
       "projectFolder": "packages/spec",
       "reviewCategory": "production",
-      "shouldPublish": false
+      "versionPolicyName": "typespec"
```
Comment (nguerrera) [inline rush.json:169]: This one isn't published, I don't think there's any need to publish it either.

Comment (nguerrera) [reply rush.json:169]: Is there a way to say, bump with this policy, but don't publish?

Comment (timotheeguerin) [reply rush.json:169]: ah yes oops, if its not published it doesn't ask for changelog and bump version anyway

**Review by mikeharder — APPROVED** [review]

File: rush.json (hunk 2 above)
Comment (timotheeguerin) [reply rush.json]: reverted

**Pull request merged**